### PR TITLE
Add support for downloadonly in DNF5

### PR DESCRIPTION
**Pull request by j-mracek**
Resolve: https://github.com/rpm-software-management/dnf5/issues/537

### CODE REVIEW

**Comment by jan-kolarik** [comment]
The PR itself looks great, but I wanted to test the parameter in CLI and it is unknown. I think we are missing the `owner.opt_binds().add("downloadonly", downloadonly);` in the main config...

File: dnf5/main.cpp [hunk 1]
```diff
@@ -769,12 +769,16 @@ int main(int argc, char * argv[]) try {
...
-                           "will be performed."
-                        << std::endl;
+            if (base.get_config().get_downloadonly_option().get_value()) {
+                std::cout << "The operation will only download packages for the transaction." << std::endl;
+            } else {
+                for (const auto & tsflag : base.get_config().get_tsflags_option().get_value()) {
+                    if (tsflag == "test") {
+                        std::cout << "Test mode enabled: Only package downloads, pgp key installations and transaction "
+                                     "checks "
+                                     "will be performed."
```
Comment (jrohel) [inline dnf5/main.cpp:779]: The message is badly cut.

**Comment by jrohel** [comment]
@jan-kolarik
> I think we are missing the owner.opt_binds().add("downloadonly", downloadonly); in the main config. Can you please fix that?

I don't think so. As you can see in the source code, `downloadonly` is a "runtime only option".
`OptionBool downloadonly{false};  // runtime only option`

And from my point of view, `downloadonly` is a specific option of dnf5 application. It is not a libdnf(5) option. Do we want to add this option to the "dnf.conf " file?

**Comment by jan-kolarik** [comment]
> @jan-kolarik
> 
> > I think we are missing the owner.opt_binds().add("downloadonly", downloadonly); in the main config. Can you please fix that?
> 
> I don't think so. As you can see in the source code, `downloadonly` is a "runtime only option". `OptionBool downloadonly{false}; // runtime only option`
> 
> And from my point of view, `downloadonly` is a specific option of dnf5 application. It is not a libdnf(5) option. Do we want to add this option to the "dnf.conf " file?

I see. I was confused that the `--downloadonly` parameter was not recognized for me when testing the PR and I just noticed the binding is missing and it might be the cause. Anyway, I tried to pull the PR again today and everything seems working correctly, so it was just probably my broken local setup.

**Review by jan-kolarik — APPROVED** [review]

**Comment by j-mracek** [comment]
> > @jan-kolarik
> > > I think we are missing the owner.opt_binds().add("downloadonly", downloadonly); in the main config. Can you please fix that?
> > 
> > 
> > I don't think so. As you can see in the source code, `downloadonly` is a "runtime only option". `OptionBool downloadonly{false}; // runtime only option`
> > And from my point of view, `downloadonly` is a specific option of dnf5 application. It is not a libdnf(5) option. Do we want to add this option to the "dnf.conf " file?
> 
> I see. I was confused that the `--downloadonly` parameter was not recognized for me when testing the PR and I just noticed the binding is missing and it might be the cause. Anyway, I tried to pull the PR again today and everything seems working correctly, so it was just probably my broken local setup.

In the first version I enabled binding to allow `--setopt=downloadonly=true` but I was not sure whether we want to support that. Be honest I did not found any reason to not add it also.

File: dnf5/main.cpp (hunk 1 above)
Comment (jan-kolarik) [reply dnf5/main.cpp:779]: It looks weird, but static tests are not complaining and also the actual output looks OK, so I don't consider this as a problem.

**Pull request merged**